packages/schematics/angular/migrations/update-17/use-application-builder.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { workspaces } from '@angular-devkit/core';
+import {
+  Rule,
+  SchematicContext,
+  SchematicsException,
+  chain,
+  externalSchematic,
+} from '@angular-devkit/schematics';
+import { JSONFile } from '../../utility/json-file';
+import { TreeWorkspaceHost, allTargetOptions, getWorkspace } from '../../utility/workspace';
+import { Builders, ProjectType } from '../../utility/workspace-models';
+
+export default function (): Rule {
+  return async (tree, context) => {
+    const rules: Rule[] = [];
+    const workspace = await getWorkspace(tree);
+
+    for (const [name, project] of workspace.projects) {
+      if (project.extensions.projectType !== ProjectType.Application) {
+        // Only interested in application projects since these changes only effects application builders
+        continue;
+      }
+
+      const buildTarget = project.targets.get('build');
+      if (!buildTarget || buildTarget.builder === Builders.Application) {
+        continue;
+      }
+
+      if (
+        buildTarget.builder !== Builders.BrowserEsbuild &&
+        buildTarget.builder !== Builders.Browser
+      ) {
+        context.logger.error(
+          `Cannot update project "${name}" to use the application builder` +
+            `as it uses "${buildTarget.builder}" which is not supported by the Angular team.`,
+        );
+
+        continue;
+      }
+
+      // Update builder target and options
+      buildTarget.builder = Builders.Application;
+      const hasServerTarget = project.targets.has('server');
+
+      for (const [, options] of allTargetOptions(buildTarget, false)) {
+        // Show warnings for using no longer supported options
+        if (usesNoLongerSupportedOptions(options, context, name)) {
+          continue;
+        }
+
+        // Rename and transform options
+        options['browser'] = options['main'];
+        if (hasServerTarget) {
+          options['server'] = (options['browser'] as string | undefined)?.replace(
+            'main.ts',
+            'main.server.ts',
+          );
```

Comment:
```suggestion
          options['server'] =  'main.server.ts';
```

Reply:
This will not be correct for apps inside of the projects dir

Reply:
I suppose this is uncommon but the browser entry could technically not contain `main.ts`.
Maybe take the directory name and add `main.server.ts` to it to be safe?